src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -510,23 +483,23 @@ public static Path getDefaultModulePath() {
     }
 
     /*
-     * Returns a module finder of the given module path or the system modules
-     * if the module path is empty that limits the observable modules to those
-     * in the transitive closure of the modules specified in {@code limitMods}
-     * plus other modules specified in the {@code roots} set.
+     * Returns a module finder of the given module finder that limits the
+     * observable modules to those in the transitive closure of the modules
+     * specified in {@code limitMods} plus other modules specified in the
+     * {@code roots} set.
      *
      * @throws IllegalArgumentException if java.base module is present
-     * but its descriptor has no version
+     * but its descriptor has no version or the java.base version is not the
+     * same as the current runtime.
      */
-    public static ModuleFinder newModuleFinder(List<Path> paths,
+    public static ModuleFinder newModuleFinder(ModuleFinder original,
                                                Set<String> limitMods,
-                                               Set<String> roots)
+                                               Set<String> roots,
+                                               boolean isRuntimeLink)
```

Comment:
`isRuntimeLink` is strictly for performance but this method has no relationship with linking from packaged modules or run-time image.   Suggest to drop this new parameter and refactor the version check as a separate method and call it before creating `JlinkConfiguration`

Reply:
Thanks. I've moved the version check outside this method.